packages/admin/cms-admin/src/generator/future/utils/tsMorphHelper.ts

```diff
@@ -0,0 +1,205 @@
+import {
+    type ArrayLiteralExpression,
+    type ArrowFunction,
+    type ExportedDeclarations,
+    type Expression,
+    type Identifier,
+    type Node,
+    type NumericLiteral,
+    type ObjectLiteralExpression,
+    Project,
+    type PropertyAssignment,
+    type SourceFile,
+    type StringLiteral,
+    SyntaxKind,
+    type VariableDeclaration,
+} from "ts-morph";
+
+import { type GeneratorConfig } from "../generator";
+
+const project = new Project({
+    tsConfigFilePath: "tsconfig.json",
+    skipAddingFilesFromTsConfig: true,
+});
+export function morphTsSource(path: string) {
+    let tsSource = project.getSourceFile(path);
+    if (!tsSource) tsSource = project.addSourceFileAtPath(path);
+    return tsSource;
+}
+
+const supportedImportPaths = [
```

Comment:
I think this is a good solution to the problem 👍🏼

Reply:
yes, it is pretty simple. But it doesn't support recursions (eg. for fieldsets)